presto-native-execution/presto_cpp/presto_protocol/ConnectorProtocol.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "ConnectorProtocol.h"
+#include "presto_protocol.h"
+
+namespace facebook::presto::protocol {
+
+namespace {
+std::unordered_map<std::string, std::unique_ptr<ConnectorProtocol>>&
+protocols() {
+  static std::unordered_map<std::string, std::unique_ptr<ConnectorProtocol>>
+      protocols;
+  return protocols;
+}
+} // namespace
+
+void registerConnectorProtocol(
+    const std::string& connectorName,
+    std::unique_ptr<ConnectorProtocol> protocol) {
+  bool ok = protocols().insert({connectorName, std::move(protocol)}).second;
+  VELOX_CHECK(
```

Comment:
Nit : Can you inline `protocols().insert({connectorName, std::move(protocol)}).second ` in the VELOX_CHECK ?